libvirtcollector/common.go

```diff
@@ -65,3 +78,74 @@ func copyNamespaceElements(ns []plugin.NamespaceElement) []plugin.NamespaceEleme
 	copy(newNs, ns)
 	return newNs
 }
+
+func removeDuplicates(elements []plugin.Metric) []plugin.Metric {
+	// Use map to record duplicates as we find them.
+	encountered := map[string]bool{}
+	result := []plugin.Metric{}
+
+	for _, v := range elements {
+		ns := joinNamespace(v.Namespace.Strings())
+		if !encountered[ns] {
+			encountered[ns] = true
+			result = append(result, v)
+		}
+	}
+	// Return the new slice.
+	return result
+}
+
+func metricStored(elements []plugin.Metric, newNamespace []plugin.NamespaceElement) bool {
+	for _, v := range elements {
+		ns := joinNamespace(v.Namespace.Strings())
```

Comment:
instead of implement function You can use strings.Join(v.Namespace.Strings(), "/")